### ISSUE TITLE
twisted.web.server.Request.childLink is buggy on requests where URL ends with a slash.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7612]
|<img alt="Tv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Tv reported|
|-|-|
|Trac ID|trac#290|
|Type|defect|
|Created|2003-09-28 00:32:31Z|

<details><summary>Searchable metadata</summary>

```
trac-id__290 290
type__defect defect
reporter__Tv Tv
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1064709151000000 1064709151000000
changetime__1470831651032356 1470831651032356
version__ 
owner__ 
cc__Tv
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Tv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Tv commented|
|-|-|

```
#!html
<pre>
I have a web resource that has isLeaf=1. URL is
http://localhost/myresource/. myresource does
request.childLink('foo'). Because request.prepath ==
['myresource', ''], childLink returns
self.prepath[-1]+'/'+name, which is ''+'/'+name = '/'+name =
'/foo', which isn't myresource's child at all.

This patch makes the problem go away, but might have bad
side effects.

diff -u -u -r1.102 server.py
--- server.py   12 Sep 2003 10:30:11 -0000      1.102
+++ server.py   27 Sep 2003 18:29:11 -0000
@@ -136,7 +136,7 @@
             return name
         else: # lpp == 0
             if len(self.prepath):
-                return self.prepath[-1] + '/' + name
+                return (self.prepath[-1] or '.') + '/' + name
             else:
                 return name

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Apparently moshez fixed this.

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub commented|
|-|-|

In changeset b348b5956dec294628dfe2400b13d3fa95acb7af
```
#!CommitTicketReference repository="" revision="b348b5956dec294628dfe2400b13d3fa95acb7af"
Merge pull request #290 from twisted/8558-lukasa-missingstreamerror: Resolve MissingStreamError on WINDOW_UPDATE

Author: Lukasa
Reviewer: adiroiban
Fixes: #8558
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

In changeset eb157b84b1c0a0f9ed97436342f9582da109c4d3
```
#!CommitTicketReference repository="" revision="eb157b84b1c0a0f9ed97436342f9582da109c4d3"
Merge pull request #290 from twisted/8558-lukasa-missingstreamerror: Resolve MissingStreamError on WINDOW_UPDATE

Author: Lukasa
Reviewer: adiroiban
Fixes: #8558
```
